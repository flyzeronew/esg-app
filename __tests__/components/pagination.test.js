import { render, screen } from '@testing-library/react'
import Pagination from './../../comps/pagination/Pagination';

const paginationData = {
    uri: '/view/trend',
    pageCount:4
}
// jest.mock('next/router', () => ({
//     useRouter: jest.fn(),
//   }));
describe("Pagination component", () => {
  const originalEnvData = process.env;
  const useRouter = jest.spyOn(require("next/router"), "useRouter");
    beforeEach(()=>{
      process.env = {
          ...originalEnvData
      }
  })
    it('render the pagination', ()=>{
        useRouter.mockImplementation(() => ({
            route: "/view/trend",
            pathname: "/view/trend",
            query: "",
            asPath: "",
          }));
        render(<Pagination uri={paginationData.uri} pageCount={paginationData.pageCount}/>);
        const pageNumberWithOne = screen.getAllByText('1');
        expect(pageNumberWithOne[1]).toHaveClass('act');
        expect(screen.getByText('2')).toBeInTheDocument();
        expect(screen.getByText('3')).toBeInTheDocument();

        expect(screen.queryAllByText('..')[1].closest('div')).toHaveClass('last');

        expect(screen.getByText('4')).toBeInTheDocument();
        expect(screen.getByText('›')).toBeInTheDocument();
    })
    it('render pagination and matchsnapshot', ()=>{
        useRouter.mockImplementation(() => ({
            route: "/view/trend",
            pathname: "/view/trend",
            query: "",
            asPath: "",
          }));
       const {container} = render(<Pagination uri={paginationData.uri} pageCount={10}/>);
       expect(container).toMatchSnapshot();
    })
    it('generates correct links', () => {
      useRouter.mockImplementation(() => ({
        query: { page: '3' },
      }));
  
      render(<Pagination uri="/view/trend" pageCount={5} />);
  
      expect(screen.getByText('1')).toHaveAttribute('href', 'https://esg-frontend-staging.tvbs.com.tw/view/trend');
      expect(screen.getByText('2')).toHaveAttribute('href', 'https://esg-frontend-staging.tvbs.com.tw/view/trend?page=2');
      expect(screen.getByText('4')).toHaveAttribute('href', 'https://esg-frontend-staging.tvbs.com.tw/view/trend?page=4');
      expect(screen.getByText('5')).toHaveAttribute('href', 'https://esg-frontend-staging.tvbs.com.tw/view/trend?page=5');
      expect(screen.getByText('‹')).toHaveAttribute('href', 'https://esg-frontend-staging.tvbs.com.tw/view/trend?page=2');
      expect(screen.getByText('›')).toHaveAttribute('href', 'https://esg-frontend-staging.tvbs.com.tw/view/trend?page=4');
    });
    it('hides previous arrow on first page', () => {
        useRouter.mockImplementation(() => ({
          query: { page: '1' },
        }));
    
        render(<Pagination uri="/view/trend" pageCount={5} />);
    
        expect(screen.queryByText('‹').closest('div')).toHaveClass('btn hide');
      });
    
      it('shows both arrow marks for middle pages', () => {
        useRouter.mockImplementation(() => ({
          query: { page: '3' },
        }));
    
        render(<Pagination uri="/view/trend" pageCount={5} />);
    
        expect(screen.getByText('1')).toBeInTheDocument();
        expect(screen.getByText('2')).toBeInTheDocument();
        expect(screen.getByText('3')).toHaveClass('act');
        expect(screen.getByText('4')).toBeInTheDocument();
        expect(screen.getByText('5')).toBeInTheDocument();
        expect(screen.getByText('‹')).toBeInTheDocument();
        expect(screen.getByText('›')).toBeInTheDocument();
      });
    
      it('shows only one arrow  for last page', () => {
        useRouter.mockImplementation(() => ({
          query: { page: '5' },
        }));
    
         render(<Pagination uri="/view/trend" pageCount={5} />);
        // screen.debug()
        expect(screen.getByText('1')).toBeInTheDocument();
        expect(screen.getByText('3')).toBeInTheDocument();
        expect(screen.getByText('4')).toBeInTheDocument();
        expect(screen.getByText('5')).toHaveClass('act');
        expect(screen.queryByText('‹')).toBeInTheDocument();
        expect(screen.queryByText('›')).not.toBeInTheDocument();
      });
})