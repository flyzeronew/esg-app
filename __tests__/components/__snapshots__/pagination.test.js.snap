// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component render pagination and matchsnapshot 1`] = `
<div>
  <div
    class="pageJump"
  >
    <div
      class="btn hide"
    >
      <a
        href="https://esg-frontend-staging.tvbs.com.tw/view/trend?page=0"
      >
        ‹
      </a>
    </div>
    <div
      class="first hide"
    >
      <a
        href="https://esg-frontend-staging.tvbs.com.tw/view/trend"
      >
        1
      </a>
      <span>
        ..
      </span>
    </div>
    <div
      class="box"
    >
      <a
        class="act"
        href=""
      >
        1
      </a>
      <a
        class=""
        href="https://esg-frontend-staging.tvbs.com.tw/view/trend?page=2"
      >
        2
      </a>
      <a
        class=""
        href="https://esg-frontend-staging.tvbs.com.tw/view/trend?page=3"
      >
        3
      </a>
    </div>
    <div
      class="last"
    >
      <span>
        ..
      </span>
      <a
        href="https://esg-frontend-staging.tvbs.com.tw/view/trend?page=10"
      >
        10
      </a>
    </div>
    <div
      class="btn"
    >
      <a
        href="https://esg-frontend-staging.tvbs.com.tw/view/trend?page=2"
      >
        ›
      </a>
    </div>
  </div>
</div>
`;
